File: app_escola_conectadaon/lib/models/resposta.dart
import 'comunicado.dart';
import 'responsavel.dart';

class Resposta {
  int? id;
  String mensagemResposta;
  Comunicado? comunicado;
  Responsavel? responsavel;
  Resposta(
      {this.id,
      required this.mensagemResposta,
      this.comunicado,
      this.responsavel});
  Map<String, dynamic> toJson() => {
        'id': id,
        'mensagemResposta': mensagemResposta,
        'comunicado': comunicado?.toJson(),
        'responsavel': responsavel?.toJson()
      };
}

File: api-escola-conectada/src/main/java/br/edu/escolaconectada/model/Resposta.java
package br.edu.escolaconectada.model;
import jakarta.persistence.*; import java.time.LocalDateTime;
@Entity
public class Resposta {
 @Id @GeneratedValue(strategy = GenerationType.IDENTITY) private Long id;
 @Column(length=2000) private String mensagemResposta; private LocalDateTime dataResposta = LocalDateTime.now();
 @ManyToOne private Comunicado comunicado; @ManyToOne private Responsavel responsavel;
 public Long getId(){return id;} public void setId(Long id){this.id=id;}
 public String getMensagemResposta(){return mensagemResposta;} public void setMensagemResposta(String mensagemResposta){this.mensagemResposta=mensagemResposta;}
 public LocalDateTime getDataResposta(){return dataResposta;} public void setDataResposta(LocalDateTime dataResposta){this.dataResposta=dataResposta;}
 public Comunicado getComunicado(){return comunicado;} public void setComunicado(Comunicado comunicado){this.comunicado=comunicado;}
 public Responsavel getResponsavel(){return responsavel;} public void setResponsavel(Responsavel responsavel){this.responsavel=responsavel;}
}
